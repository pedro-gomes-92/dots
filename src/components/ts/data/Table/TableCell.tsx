import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { Base, BaseProps } from '../../Base';
import { Text } from '../../text';

export interface TableCellProps extends BaseProps {
  text: string;
}

export const TableCell: FunctionComponent<TableCellProps> = ({
  className,
  text,
  ...rest
}: TableCellProps): JSX.Element => {
  return (
    <Base className={classnames('table-cell', className)} {...rest}>
      <Text text={text} />
    </Base>
  );
};

TableCell.defaultProps = {
  tag: 'td',
};
