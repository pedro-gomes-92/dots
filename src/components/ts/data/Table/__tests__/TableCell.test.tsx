import React from 'react';
import { render } from '@testing-library/react';

import { TableCell } from '../TableCell';

describe('Given <TableRow />', (): void => {
  const addTableHTML = (content: JSX.Element): JSX.Element => (
    <table>
      <tbody>
        <tr>{content}</tr>
      </tbody>
    </table>
  );

  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      const { container } = render(addTableHTML(<TableCell text="" />));
      expect(container).toMatchSnapshot();
    });

    describe('And uses text', (): void => {
      it('Then should render text value', (): void => {
        const { container } = render(addTableHTML(<TableCell text="a" />));
        expect(container.querySelector('.text').textContent).toEqual('a');
      });
    });
  });
});
