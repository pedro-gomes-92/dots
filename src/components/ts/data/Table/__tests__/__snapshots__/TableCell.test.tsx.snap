// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <TableRow /> When initializes Then should match the snapshot 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="table-cell"
        >
          <span
            class="text"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
